src/signing/interactors/batch_tx_batch_key_signing_request.rs

```diff
@@ -0,0 +1,68 @@
+use crate::prelude::*;
+
+/// A batch of keys (derivation paths) all being factor instances of a HDFactorSource
+/// with id `factor_source_id` to sign a single transaction with, which hash
+/// is `intent_hash`.
+#[derive(PartialEq, Eq, Clone, Debug, Hash)]
+pub struct BatchKeySigningRequest {
+    /// Hash to sign
+    intent_hash: IntentHash,
+
+    /// ID of factor to use to sign
+    pub factor_source_id: FactorSourceIDFromHash,
```

Comment:
Seems redundant if the hosts will receive `BatchTXBatchKeySigningRequest` which already contains `factor_source_id`

Reply:
does not cost anything and is more beautiful than `owned_factor_instances.first().unwrap().factor_source_id` :D 

it is a more canonical representation.

Reply:
> owned_factor_instances.first().unwrap().factor_source_id

In which context? Isn't  `BatchKeySigningRequest` always wrapped in `BatchTXBatchKeySigningRequest`?

I mostly ask because we have `factor_source_id` defined in `BatchTXBatchKeySigningRequest`, `BatchKeySigningRequest` and then per each `OwnedFactorInstance`, all o them should actually be the same, while the types potentially allow to have the inconsistency. If it is not possible to define the `factor_source_id` only once in the `BatchTXBatchKeySigningRequest` hierarchy, then I am ok with your current setup.

Reply:
TODO try clean up